src/olympia/search/filters.py

```diff
@@ -212,20 +213,19 @@ def get_values(self):
                         'Invalid Return To AMO guid (not in base64url format?)'
                     )
                 )
-
-            # Unfortunately we have to check against the database here. We only
-            # need to check that a DiscoveryItem exists, if somehow the add-on
-            # is not public, it will get filtered out later by
-            # ReviewedContentFilter.
-            if not DiscoveryItem.objects.filter(addon__guid=value).exists():
-                raise ValueError(
-                    ugettext(
-                        'Invalid Return To AMO guid (not a curated add-on)'
-                    )
-                )
-            return [value] if value else []
+            # Filter on the now decoded guid param as normal, then add promoted
+            # filter on top to only return "safe" add-ons for return to AMO.
+            # We don't care about the app param - we just want to ensure the
+            # add-ons are "safe".
+            filters = super().get_es_query([value])
```

Comment:
mmm, did leaving all the decoding `rta:...` logic in `get_values` not work?

Reply:
It did, but `get_values()` cannot return the additional filter for promoted add-ons, so I would have had to implement that elsewhere anyway. I figure I might as well override the whole `get_es_query()` then. I'm open to better ways of doing this, didn't want to refactor too far without a better plan...

Reply:
I don't know if there's a "more correct" way, I just saw you had to change `AddonQueryMultiParam.get_es_query` to make it work because the code was moved from `get_values` to this method and I was curious why.

Reply:
Ah yeah, I still had to modify the `get_es_query()` method to allow an arbitrary value to be passed.

One alternative way of doing this would be to override `get_values()` to do the decoding, store a flag indicating that we had to decode something, and add the promoted condition in `get_es_query()` if that flag is present. I thought it wasn't that much better.